keanu-project/src/main/java/io/improbable/keanu/vertices/NotObservable.java

```diff
@@ -0,0 +1,20 @@
+package io.improbable.keanu.vertices;
+
+public class NotObservable<T> implements Observable<T> {
+    // package private
+    NotObservable() {}
+
+    @Override
+    public void observe(T value) {
+        throw new UnsupportedOperationException("This type of vertex does not support being observed");
+    }
+
+    @Override
+    public void unobserve() {
```

Comment:
Yes - but agreed that's not very informative so I've added this:

        // Do nothing: 
        // This is allowed, since it's common for algorithms to unobserve all vertices
        // e.g. in io.improbable.keanu.network.NetworkSnapshot.apply

Reply:
Nice